index.d.ts

```diff
@@ -471,6 +499,36 @@ declare namespace Moleculer {
 		}
 	}
 
+	class Transit {
+		constructor(broker: ServiceBroker, transporter: Transporter, opts: GenericObject);
+		afterConnect(wasReconnect: boolean): Promise<void>;
+		connect(): Promise<void>;
+		disconnect(): Promise<void>;
+		sendDisconnectPacket(): Promise<void>;
+		makeSubscriptions(): Promise<Array<void>>;
+		messageHandler(cmd: string, msg: GenericObject): boolean | Promise<void> | undefined;
+		request(ctx: Context): Promise<void>;
+		sendEvent(nodeID: string, eventName: string, data: GenericObject): void;
+		sendBalancedEvent(eventName: string, data: GenericObject, nodeGroups: GenericObject): void;
+		sendEventToGroups(eventName: string, data: GenericObject, groups: Array<any>): void;
+		removePendingRequest(id: string): void
+		removePendingRequestByNodeID(nodeID: string): void;
+		sendResponse(nodeID: string, id: string, data: GenericObject, err: Error): Promise<void>;
+		sendResponse(nodeID: string, id: string, data: GenericObject): Promise<void>;
+		discoverNodes(): Promise<void>;
+		discoverNode(nodeID: string): Promise<void>;
+		sendNodeInfo(nodeID: string): Promise<void | Array<void>>;
+		sendPing(nodeID: string): Promise<void>;
+		sendPong(payload: GenericObject): Promise<void>;
+		processPong(payload: GenericObject): void;
+		sendHeartbeat(localNode: NodeHealthStatus): Promise<void>;
+		subscribe(topic: string, nodeID: string): Promise<void>;
+		publish(packet: Packet): Promise<void>;
+		serialize(obj: GenericObject, type: string): Buffer;
+		deserialize(buf: Buffer, type: string): any
+		deserialize(buf: Buffer): any
+	}
+
```

Comment:
The Transit is an internal class, so you don't call them directly. So maybe it is not neccessary to publish in d.ts.

Reply:
I noticed that it can be used while creating a custom transport but perhaps an interface instead so it can't be instantiated.

Reply:
It's true, transporter calls directly the transit.